src/Workspaces/VisualBasic/Portable/LanguageServices/VisualBasicSemanticFactsService.vb

```diff
@@ -145,16 +145,26 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         Public Function GetDeclaredSymbol(semanticModel As SemanticModel, token As SyntaxToken, cancellationToken As CancellationToken) As ISymbol Implements ISemanticFactsService.GetDeclaredSymbol
             Dim location = token.GetLocation()
 
-            Dim q = From node In token.GetAncestors(Of SyntaxNode)()
-                    Where Not TypeOf node Is AggregationRangeVariableSyntax AndAlso
-                          Not TypeOf node Is CollectionRangeVariableSyntax AndAlso
-                          Not TypeOf node Is ExpressionRangeVariableSyntax AndAlso
-                          Not TypeOf node Is InferredFieldInitializerSyntax
-                    Let symbol = semanticModel.GetDeclaredSymbol(node, cancellationToken)
-                    Where symbol IsNot Nothing AndAlso symbol.Locations.Contains(location)
-                    Select symbol
-
-            Return q.FirstOrDefault()
+            For Each ancestor In token.GetAncestors(Of SyntaxNode)()
+                If Not TypeOf ancestor Is AggregationRangeVariableSyntax AndAlso
+                   Not TypeOf ancestor Is CollectionRangeVariableSyntax AndAlso
+                   Not TypeOf ancestor Is ExpressionRangeVariableSyntax AndAlso
+                   Not TypeOf ancestor Is InferredFieldInitializerSyntax Then
+
+                    Dim symbol = semanticModel.GetDeclaredSymbol(ancestor)
+
+                    If symbol IsNot Nothing Then
+                        If symbol.Locations.Contains(location) Then
+                            Return symbol
+                        End If
+
+                        ' We found some symbol, but it defined something else. We're not going to have a higher node defining _another_ symbol with this token, so we can stop now.
+                        Return Nothing
+                    End If
+                End If
```

Comment:
why no statement check here?

Reply:
Oops forgot to comment: statements in VB are a funny beast. Unlike C# where it's really an executable statement, a VB statement is just loosely "something that is on it's own line kinda somewhat." So a MethodBlockSyntax has a MethodStatementSyntax, and that is the thing that has the token. So stopping at any arbitrary statement would be problematic.

Reply:
Gotcha.  Note: we have a helper for this with ISyntaxFactsService.IsExecutableStatement.

Reply:
Or you can just check for ExecutableStatementSyntax at the VB level.

Reply:
Ah I'll update then.

Reply:
Fixed, and on both sides now using the SyntaxFactsService method so if we ever change that we don't have to change both.